x-pack/plugins/enterprise_search/public/applications/app_search/components/settings/log_retention/generic_confirmation_modal.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ReactNode, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFormRow,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+
+interface GenericConfirmationModalProps {
+  description: ReactNode;
+  subheading: ReactNode;
+  target: string;
+  title: ReactNode;
+  onClose(): void;
+  onSave(): void;
+}
+
+export const GenericConfirmationModal: React.FC<GenericConfirmationModalProps> = ({
+  description,
+  onClose,
+  onSave,
+  subheading,
+  target,
+  title,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+
+  const onConfirm = () => {
+    setInputValue('');
+    onSave();
+  };
+
+  return (
+    <EuiModal onClose={onClose}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>{title}</EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiText>
+          <p>
+            <strong>{subheading}</strong>
+          </p>
+          <p>{description}</p>
+        </EuiText>
+        <EuiSpacer />
+        <EuiFormRow
+          label={i18n.translate(
+            'xpack.enterpriseSearch.appSearch.settings.logRetention.modal.prompt',
+            {
+              defaultMessage: 'Type "{target}" to confirm.',
+              values: { target },
+            }
+          )}
+        >
+          <EuiFieldText
+            data-test-subj="GenericConfirmationModalInput"
+            name="engineName"
+            value={inputValue}
+            onChange={(e) => setInputValue(e.target.value)}
+          />
+        </EuiFormRow>
+      </EuiModalBody>
+      <EuiModalFooter>
+        <EuiButtonEmpty data-test-subj="GenericConfirmationModalCancel" onClick={onClose}>
+          {i18n.translate('xpack.enterpriseSearch.appSearch.settings.logRetention.modal.cancel', {
+            defaultMessage: 'Cancel',
+          })}
+        </EuiButtonEmpty>
+        <EuiButton
+          data-test-subj="GenericConfirmationModalSave"
+          onClick={onConfirm}
+          disabled={inputValue !== target}
```

Comment:
Another accessibility request - can we change this to normalize casing?

```suggestion
          disabled={inputValue.toLowerCase() !== target.toLowerCase()}
```

There's almost virtually no way for a non-sighted user to know DISABLE is in all caps (some screen readers may enunciate it more but that's still not a great tell). And honestly it's not like a user somehow didn't mean it if they typed in uppercase vs lowercase disable and then clicked 'save' anyway.

Reply:
Yeah, I noticed the same thing. This is a great suggestion.

Reply:
https://github.com/elastic/kibana/pull/83009/commits/e22098287934276b58bcd613f72241502e6b3b38